Ensembl Variant Effect Predictor annotations support

I have already been using Adam with snpEff annotations and it worked great. However, when I've tried to use it with a VCF file annotated using Ensembl VEP it didn't work. I was expecting, that ANN format is standardized enough, so that ```transcriptEffects``` will be populated.

So I wanted to ask whether I do something wrong and there is a way to make Adam work with VEP or it's not yet supported?

Thank you.

Dear @heuermh ,

thank you for an answer.
I checked cannoli way of running VEP and I've tried to do it myself.

I've used a very simple VCF file, containing one change:
```
chr1    2309937 .       C       CTT     281.04  .       AC=2;AF=1;AN=2   GT:AD:DP:GQ:PL  1/1:0,8:8:24:295,24,0
```
I've annotated it with VEP, release 96:
```
vep         --dir /opt/vep/.vep/         -i /data/input.vcf.gz -o /data/output2.vep.vcf --format vcf --vcf --vcf_info_field ANN --terms SO --no_stats --offline
```
Annotated VCF had lines:

```
##INFO=<ID=ANN,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID">
(...)
chr1    2309937 .       C       CTT     281.04  .       AC=2;AF=1;AN=2;DP=18;ExcessHet=3.0103;FS=0;MLEAC=2;MLEAF=1;MQ=59.4;QD=28.73;SOR=1.863;ANN=TT|3_prime_UTR_variant|MODIFIER|SKI|ENSG00000157933|Transcript|ENST00000378536|protein_coding|7/7||||5807-5808|||||||1||HGNC|HGNC:10896       GT:AD:DP:GQ:PL  1/1:0,8:8:24:295,24,0
```

But now running:
```
ac.loadVcf("/tmp/output2.vep.vcf").toVariants.dataset.show
```
throws:
```
Caused by: java.lang.NumberFormatException: For input string: "5807-5808"
  at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
  at java.lang.Integer.parseInt(Integer.java:580)
  at java.lang.Integer.parseInt(Integer.java:615)
  at org.bdgenomics.adam.converters.TranscriptEffectConverter$.parseFraction(TranscriptEffectConverter.scala:99)
  at org.bdgenomics.adam.converters.TranscriptEffectConverter$.parseTranscriptEffect(TranscriptEffectConverter.scala:143)
```
as ```5807-5808``` is not a number. 

Is there anything that I'm doing wrong?

--- a/adam-core/src/main/scala/org/bdgenomics/adam/converters/TranscriptEffectConverter.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/converters/TranscriptEffectConverter.scala
@@ -84,6 +84,18 @@ private[adam] object TranscriptEffectConverter extends Serializable with Logging
   }
 
   /**
+   * Ensembl VEP incorrectly supplies an interval instead of a position
+   * for some attributes; in these cases use the interval start as the position.
+   *
+   * @param s position or interval
+   * @return position or interval start
+   */
+  private def positionOrIntervalStart(s: String): Int = {
+    val tokens = s.split("-")
+    Integer.parseInt(tokens(0))
+  }
+
+  /**
    * Split a single or fractional value into optional numerator and denominator values.
    *
    * @param s single or fractional value to split
@@ -96,8 +108,8 @@ private[adam] object TranscriptEffectConverter extends Serializable with Logging
     val tokens = s.split("/")
     tokens.length match {
       case 0 => (None, None)
-      case 1 => (Some(Integer.parseInt(tokens(0))), None)
-      case _ => (Some(Integer.parseInt(tokens(0))), Some(Integer.parseInt(tokens(1))))
+      case 1 => (Some(positionOrIntervalStart(tokens(0))), None)
+      case _ => (Some(positionOrIntervalStart(tokens(0))), Some(Integer.parseInt(tokens(1))))
     }
   }
 

--- a/adam-core/src/test/scala/org/bdgenomics/adam/converters/TranscriptEffectConverterSuite.scala
+++ b/adam-core/src/test/scala/org/bdgenomics/adam/converters/TranscriptEffectConverterSuite.scala
@@ -35,6 +35,7 @@ class TranscriptEffectConverterSuite extends ADAMFunSuite {
   final val INVALID_NUMBER = "T|upstream_gene_variant||TAS1R3|ENSG00000169962|transcript|ENST00000339381.5|protein_coding|1/2|c.-485C>T|||4|1/42|not a number|"
   final val INVALID_FRACTION = "T|upstream_gene_variant||TAS1R3|ENSG00000169962|transcript|ENST00000339381.5|protein_coding|not a number/2|c.-485C>T|||4|1/42|453|"
   final val VALID = "T|upstream_gene_variant||TAS1R3|ENSG00000169962|transcript|ENST00000339381.5|protein_coding|1/2|c.-485C>T|||4|1/42|453|"
+  final val VEP_POSITION = "T|upstream_gene_variant||TAS1R3|ENSG00000169962|transcript|ENST00000339381.5|protein_coding|1/2|c.-485C>T|||4-5/420|1/42|453|"
   final val DIFFERENT_ALT = "A|upstream_gene_variant||TAS1R3|ENSG00000169962|transcript|ENST00000339381.5|protein_coding|1/2|c.-485C>T|||4|1/42|453|"
 
   var variant: Variant = null
@@ -168,6 +169,16 @@ class TranscriptEffectConverterSuite extends ADAMFunSuite {
       assert(te.getDistance == 453)
       assert(te.getMessages.isEmpty)
     })
+  }
+
+  test("parse VCF ANN attribute with Ensembl VEP position attribute") {
+    val ann = TranscriptEffectConverter.parseAnn(Seq(VEP_POSITION), ValidationStringency.STRICT)
+    assert(ann.length == 1)
+
+    val te = ann.head
+    assert(te.getAlternateAllele == "T")
+    assert(te.getCdsPosition == 4)
+    assert(te.getCdsLength == 420)
   }
 
   test("convert to transcript effect from null VCF ANN attribute in variant context") {

